subprojects/base-services/src/main/java/org/gradle/internal/reflect/DirectInstantiator.java

```diff
@@ -136,4 +136,11 @@ private boolean isMatch(Class<?>[] argumentTypes, Class[] parameterTypes) {
             return true;
         }
     }
+
+    public static class CachedConstructor extends ReflectionCache.CachedInvokable<Constructor<?>> {
```

Comment:
Should be `private`.

Reply:
Should `CachedInvokable` be a top-level class?

Reply:
`CachedConstructor` is now private and `CachedInvokable` a top-level class.